packages/vite/src/node/plugins/clientInjections.ts

```diff
@@ -47,7 +50,10 @@ export function clientInjectionsPlugin(config: ResolvedConfig): Plugin {
           .replace(`__DEFINES__`, serializeDefine(config.define || {}))
           .replace(`__HMR_PROTOCOL__`, JSON.stringify(protocol))
           .replace(`__HMR_HOSTNAME__`, JSON.stringify(host))
-          .replace(`__HMR_PORT__`, JSON.stringify(port))
+          .replace(
+            `__HMR_PORT__`,
+            options.runtimePort ? options.runtimePort : JSON.stringify(port)
```

Comment:
**nitpick:** I think this can be shortcutted to `options.runtimePort ?? JSON.stringify(port)`

Reply:
But by the way, do we need to use `JSON.stringify(options.runtimePort ?? port)` here instead?

Reply:
I have made the shortcut in the lattest commit.

If we change the line to `JSON.stringify(options.runtimePort ?? port)` , 
runtimePort 
e.g. `'window.location.href'` would become `const socketHost = ${__HMR_HOSTNAME__ || location.hostname}:${"window.location.href"}` in the client code, 
but what we expect is `const socketHost = ${__HMR_HOSTNAME__ || location.hostname}:${window.location.href}`